test/volumes-create-name.test.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2017, Joyent, Inc.
+ */
+
+var assert = require('assert-plus');
+var test = require('tape').test;
+var vasync = require('vasync');
+
+var common = require('./common');
+var mod_config = require('../lib/config.js');
+var mod_testConfig = require('./lib/config');
+var mod_testNetworks = require('./lib/networks');
+var mod_testVolumes = require('./lib/volumes');
+
+var CONFIG = mod_config.configure();
+
+/*
+ * This regular expression is not meant to match the general ISO 8601 format,
+ * only the specific format outputted by new Date().toISOString().
+ */
+var ISO_DATE_STRING_RE = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z/;
+
+if (CONFIG.experimental_nfs_shared_volumes !== true) {
+    console.log('experimental_nfs_shared_volumes setting not enabled, ' +
+        'skipping tests');
+    process.exitCode = 0;
+} else {
+    var CLIENTS;
+    var CLIENT;
+    var SERVER;
+
+    var createdVolumes = [];
+    var testVolume;
+    var testVolumeStorageVmUuid;
+
+    function onVolumeCreatedCreator(t) {
+        return function onVolumeCreated(volumeCreationErr, req, res, volume) {
+            if (!volumeCreationErr && volume) {
+                createdVolumes.push(volume);
+                testVolume = volume;
+            }
+
+            t.ifErr(volumeCreationErr,
+                'creating a volume with no name should succeed');
+            t.ok(testVolume, 'should have set testVolume');
+            t.equal(testVolume.name.length, 64,
+                'returned volume should have a name, got: '
+                + JSON.stringify(testVolume.name));
+
```

Comment:
I don't know. What do you think? I thought about this only briefly and considered that the pattern was set by volapi so it's really up to it. But I'd be fine checking here too. I guess since we're checking the length, we already need to change this test if volapi changes how it generates names. Which could be a problem. Maybe I should just check that the length is > 1?